modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/DefaultExtensionsClient.java

```diff
@@ -0,0 +1,182 @@
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.client.OperationParameter;
+import org.mule.runtime.extension.api.client.OperationParameters;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.module.extension.internal.manager.ExtensionManagerAdapter;
+import org.mule.runtime.module.extension.internal.runtime.objectbuilder.DefaultObjectBuilder;
+import org.mule.runtime.module.extension.internal.runtime.operation.OperationMessageProcessor;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ValueResolver;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+import javafx.util.Pair;
+import org.reactivestreams.Publisher;
+
+/**
+ * This is the default implementation for a {@link ExtensionsClient}, it uses the {@link ExtensionManagerAdapter}
+ * in the {@link MuleContext} to search for the extension that wants to execute the operation from.
+ * <p>
+ * The concrete execution of the operation is handled by an {@link OperationMessageProcessor} instance.
+ * <p>
+ * This implementation can only execute extensions that were built using the SDK, Smart Connectors operations can't be executed.
+ *
+ * @since 4.0
+ */
+public final class DefaultExtensionsClient implements ExtensionsClient, Initialisable {
+
+  @Inject
+  private MuleContext muleContext;
+  private PolicyManager policyManager;
+  private ExtensionManagerAdapter extensionManager;
+  private final Map<Pair<String, String>, OperationModel> operations = new LinkedHashMap<>();
+
+  /**
+   * {@inheritDoc}
+   * <p>
+   * Set's up the {@link PolicyManager} and the {@link ExtensionManager} used to execute the operations.
+   */
+  @Override
+  public void initialise() throws InitialisationException {
+    this.extensionManager = (ExtensionManagerAdapter) muleContext.getExtensionManager();
+    try {
+      this.policyManager = muleContext.getRegistry().lookupObject(PolicyManager.class);
+    } catch (RegistrationException e) {
+      throw new InitialisationException(createStaticMessage("Cannot obtain Policy Manager"), e, this);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T, A extends Attributes> CompletableFuture<Result<T, A>> executeAsync(String extension, String operation,
+                                                                                OperationParameters parameters)
+    throws MuleException {
+    OperationMessageProcessor processor = createProcessor(extension, operation, parameters);
+    Publisher<Event> publisher = processor.apply(just(getInitialiserEvent(muleContext)));
+    //return from(publisher).map(e -> Result.<T, A>builder(e.getMessage()).build());
+    return null;
```

Comment:
null?  You can use `Mono.from(processor.apply(just(getInitialiserEvent(muleContext)))).toFuture()`